experimental/examples/prometheus/index.js

```diff
@@ -39,6 +39,23 @@ observableCounter.addCallback(observableResult => {
   observableResult.observe(counter, attributes);
 });
 
+const randomMetricPromise = async () =>
+  new Promise(resolve =>
+    setTimeout(resolve(Math.floor(Math.random() * 100)), 50)
+  );
+
+const observableGauge = meter.createObservableGauge(
+  'observable_gauge_requests',
+  {
+    description: 'Example of an ObservableGauge',
+  }
+);
+// Callbacks are run when metrics are scraped
+observableGauge.addCallback(async observableResult => {
+  const value = await randomMetricPromise();
+  observableResult.observe(value, attributes);
+});
```

Comment:
hmm, why not observe the value synchronously here? :thinking:

Reply:
> hmm, why not observe the value synchronously here? 🤔

Often, when I had to observe a value at querying time, it was from an async process, so I just replicated what I often deal with :)

Reply:
I agree that the ability to collect gauge asynchronously can be a good point to present in the example.